### ISSUE TITLE
XDP Support

### ISSUE THREAD

**Issue by huseyinsaatci** [cloudflare/ebpf_exporter#208]
I was using ebpf_exporter in my project and realized that there is no XDP support in the project. I forked the project and edited it to support XDP programs. I used libbpfgo's _AttachXDP_ function to attach the XDP program to the required interface.

My edit contains two new fields in YAML files: **is_xdp**  to indicate if the metric has an XDP program, and **interfaces** is an array with interfaces to attach the XDP program. And usage is like:
```
is_xdp: true
interfaces: ["enp8s0", "docker0"]
```
If you consider it worth adding support for XDP programs I can open a PR.

**Comment by bobrik** [comment]
What do you think about using `SEC("xdp/enp8s0")` instead (see: https://github.com/cloudflare/ebpf_exporter/pull/154)? The downside is that the interface is hardcoded in a program, but the upside is that it's consistent with other `SEC` setups.

In your setup I don't think there's a need for `is_xdp`. You can detect it from `SEC("xdp")`. Having `interfaces` set in the config means that you attach all programs to all interfaces. I can imagine cases where you'd want to have different programs on different interfaces as a part of the same config.

**Comment by huseyinsaatci** [comment]
Makes sense but when I try to use SEC("xdp/enp8s0") in my program I get this error: 
```
missing BPF prog type, check ELF section name 'xdp/enp8s0
failed to load: -22
Error attaching exporter: error loading bpf object from "src/xdp.bpf.o" for config "xdp": failed to load BPF object: invalid argument'
```
I don't think it recognizes `'xdp/enp8s0'` as an XDP program.
It is possible to get the interface from the SEC part but only if this problem can be solved.

**Comment by bobrik** [comment]
I linked https://github.com/cloudflare/ebpf_exporter/pull/154 because it implemented `SEC("perf_event/type=0,config=3,frequency=99")`.

You can do a similar thing here to implement `SEC("xdp/enp8s0")`.

**Comment by huseyinsaatci** [comment]
Okay, I will examine the codes and try to implement a similar one. I hope it's not a problem if I ask questions about the codes if I have problems.

**Comment by bobrik** [comment]
I merged #215, thanks!
